mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/SubscriptionStateHandlerImpl.java

```diff
@@ -81,11 +83,20 @@ protected void runOneIteration() {
 
     @Override
     public void onSinkUnsubscribed() {
+        if (currentState.get() == null) {
+            log.error("currentState in SubscriptionStateHandlerImpl is not set onSinkUnsubscribed");
+            return;
+        }
         currentState.updateAndGet(SubscriptionState::onSinkUnsubscribed);
     }
 
     @Override
     public void onSinkSubscribed() {
+        // sink subscription can happen before the startup of this service.
+        if (currentState.get() == null) {
+            currentState.compareAndSet(null, SubscriptionState.of(clock));
```

Comment:
same question, I was thinking the if-check is not necessary. `currentState.compareAndSet(null, SubscriptionState.of(clock));` works just fine.

Reply:
yeah it's not for thread safety but just a minor save to avoid the extra SubscriptionState.of(clock) creation.